### PR TITLE
currency_rate_update: fix wrong use of timetuple.

### PR DESCRIPTION
**Pull request by mathben**
Remove error "TypeError: argument must be 9-item sequence, not datetime.datetime"

### CODE REVIEW

File: currency_rate_update/currency_rate_update.py [hunk 1]
```diff
@@ -119,13 +119,14 @@ class Currency_rate_update(osv.Model):
     _name = "currency.rate.update"
     _description = "Currency Rate Update"
     # Dict that represent a cron object
+    nextcall_time = (datetime.today() + timedelta(days=1)).timetuple()
+    nextcall = time.strftime("%Y-%m-%d %H:%M:%S", nextcall_time)
```
Comment (bwrsandman) [inline currency_rate_update/currency_rate_update.py]: You should be using `DEFAULT_SERVER_DATETIME_FORMAT` here

```
from openerp.tools import DEFAULT_SERVER_DATETIME_FORMAT
```

File: currency_rate_update/currency_rate_update.py [hunk 2]
```diff
@@ -119,13 +119,14 @@ class Currency_rate_update(osv.Model):
     _name = "currency.rate.update"
     _description = "Currency Rate Update"
     # Dict that represent a cron object
+    nextcall_time = (datetime.today() + timedelta(days=1)).timetuple()
```
Comment (pedrobaeza) [inline currency_rate_update/currency_rate_update.py]: You don't even have to do this change to timetuple. Simply put this:

```
nextcall_time = datetime.today() + timedelta(days=1)
nextcall = nextcall_time.strftime(DEFAULT_SERVER_DATETIME_FORMAT)
```

**Comment by mathben** [comment]
Thanks for you review pedrobaeza and bwrsandman.

File: currency_rate_update/currency_rate_update.py [hunk 3]
```diff
@@ -244,12 +247,12 @@ def run_currency_update(self, cr, uid):
                     # Show the most recent note at the top
                     msg = "%s \n%s currency updated. %s" % \
                           (log_info or '',
-                           datetime.strftime(datetime.today(), '%Y-%m-%d %H:%M:%S'),
+                           datetime.strftime(datetime.today(), DEFAULT_SERVER_DATETIME_FORMAT),
```
Comment (bwrsandman) [inline currency_rate_update/currency_rate_update.py]: Better to have:

```
datetime.today().strftime(DEFAULT_SERVER_DATETIME_FORMAT)
```

File: currency_rate_update/currency_rate_update.py [hunk 4]
```diff
@@ -244,12 +247,12 @@ def run_currency_update(self, cr, uid):
...
                            note)
                     service.write({'note': msg})
                 except Exception as exc:
                     error_msg = "\n%s ERROR : %s %s" %\
-                                (datetime.strftime(datetime.today(), '%Y-%m-%d %H:%M:%S'),
+                                (datetime.strftime(datetime.today(), DEFAULT_SERVER_DATETIME_FORMAT),
```
Comment (bwrsandman) [inline currency_rate_update/currency_rate_update.py]: same as above

File: currency_rate_update/currency_rate_update.py [hunk 5]
```diff
@@ -397,8 +400,8 @@ def check_rate_date(self, rate_date, max_delta_days):
             )
 
         # We always have a warning when rate_date != today
-        rate_date_str = datetime.strftime(rate_date, '%Y-%m-%d')
-        if rate_date_str != datetime.strftime(datetime.today(), '%Y-%m-%d'):
+        rate_date_str = datetime.strftime(rate_date, DEFAULT_SERVER_DATE_FORMAT)
+        if rate_date_str != datetime.strftime(datetime.today(), DEFAULT_SERVER_DATE_FORMAT):
```
Comment (bwrsandman) [inline currency_rate_update/currency_rate_update.py]: Both lines could be

```
if rate_date.date() != datetime.today().date():
```

File: currency_rate_update/currency_rate_update.py [hunk 6]
```diff
@@ -219,7 +222,7 @@ def run_currency_update(self, cr, uid):
                         main_curr,
                         service.max_delta_days
                     )
-                    rate_name = time.strftime('%Y-%m-%d')
+                    rate_name = time.strftime(DEFAULT_SERVER_DATE_FORMAT)
```
Comment (bwrsandman) [inline currency_rate_update/currency_rate_update.py:225]: SyntaxError?

**Comment by eLBati** [comment]
Can't investigate further now, but I'm still getting

> TypeError: argument must be 9-item sequence, not datetime.datetime

even using https://github.com/savoirfairelinux/account-financial-tools
See https://travis-ci.org/OCA/l10n-italy/builds/30278029#L582

File: currency_rate_update/currency_rate_update.py (hunk 6 above)
Comment (mathben) [inline currency_rate_update/currency_rate_update.py:225]: seems good
DEFAULT_SERVER_DATE_FORMAT = "%Y-%m-%d"

Comment (mathben) [inline currency_rate_update/currency_rate_update.py:225]: This return by example : '2014-07-18'

Comment (bwrsandman) [inline currency_rate_update/currency_rate_update.py:225]: Right

**Comment by bwrsandman** [comment]
:+1:

**Comment by bwrsandman** [comment]
@eLBati 
You're getting the error because:
https://github.com/OCA/l10n-italy/commit/ae88cc78c6097826a58a128379b229205f4dbcbc#diff-354f30a63fb0907d4ad57269548329e3R13
Uses the 7.0 branch and the fix is on branch fix_currency_rate_update-wrong_timedelta

try using:

```
git clone https://github.com/savoirfairelinux/account-financial-tools.git $HOME/account-financial-tools -b fix_currency_rate_update-wrong_timedelta
```

**Comment by max3903** [comment]
:+1:

**Comment by eLBati** [comment]
Sorry for the noise
Il giorno 19/lug/2014 00:45, "Sandy" notifications@github.com ha scritto:

**Pull request merged**